From 8c6107feb2e8b7bca5df0d5138e3dac3244db732 Mon Sep 17 00:00:00 2001
From: Yimin Peng <yiminp@codeaurora.org>
Date: Wed, 16 Aug 2017 15:57:22 +0800
Subject: [PATCH] enable scm for pil.

revert the former workaround and restore scm/qcpe to load modem firmware.

Change-Id: Iebc13e512b01f5485d61057c143a0d162acb7add
Signed-off-by: Yimin Peng <yiminp@codeaurora.org>
---
 drivers/soc/qcom/pil-msa.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/qcom/pil-msa.c b/drivers/soc/qcom/pil-msa.c
index 5fcb0f9..ec5319c 100644
--- a/drivers/soc/qcom/pil-msa.c
+++ b/drivers/soc/qcom/pil-msa.c
@@ -576,8 +576,11 @@ int pil_mss_reset_load_mba(struct pil_desc *pil)
 	drv->mba_dp_size = SZ_1M;
 
 	arch_setup_dma_ops(dma_dev, 0, 0, NULL, 0);
-
+#ifdef CONFIG_MSM_GVM_QUIN
+	dma_dev->coherent_dma_mask = DMA_BIT_MASK(32);
+#else
 	dma_dev->coherent_dma_mask = DMA_BIT_MASK(sizeof(dma_addr_t) * 8);
+#endif
 
 	init_dma_attrs(&md->attrs_dma);
 	dma_set_attr(DMA_ATTR_SKIP_ZEROING, &md->attrs_dma);
@@ -689,7 +692,11 @@ static int pil_msa_auth_modem_mdt(struct pil_desc *pil, const u8 *metadata,
 
 
 	trace_pil_func(__func__);
+#ifdef CONFIG_MSM_GVM_QUIN
+	dma_dev->coherent_dma_mask = DMA_BIT_MASK(32);
+#else
 	dma_dev->coherent_dma_mask = DMA_BIT_MASK(sizeof(dma_addr_t) * 8);
+#endif
 	dma_set_attr(DMA_ATTR_SKIP_ZEROING, &attrs);
 	dma_set_attr(DMA_ATTR_STRONGLY_ORDERED, &attrs);
 	/* Make metadata physically contiguous and 4K aligned. */
-- 
1.9.1

